Network/Control/Flow.hs

```diff
@@ -163,25 +160,27 @@ maybeOpenRxWindow
     -> (RxFlow, Maybe Int)
     -- ^ 'Just' if the size should be informed to the peer.
 maybeOpenRxWindow consumed fct flow@RxFlow{..}
-    | available < threshold && winUpdate > minSize =
+    | winUpdate >= threshold =
         let flow' =
                 flow
-                    { rxfConsumed = consumed'
+                    { rxfConsumed = rxfConsumed'
                     , rxfLimit = rxfLimit'
                     }
             update = case fct of
                 FCTWindowUpdate -> winUpdate
                 FCTMaxData -> rxfLimit'
          in (flow', Just update)
     | otherwise =
-        let flow' = flow{rxfConsumed = consumed'}
+        let flow' = flow{rxfConsumed = rxfConsumed'}
          in (flow', Nothing)
   where
-    available = rxfLimit - rxfReceived
-    threshold = rxfBufSize `unsafeShiftR` 1
-    minSize = rxfBufSize `unsafeShiftR` 3
-    consumed' = rxfConsumed + consumed
-    rxfLimit' = consumed' + rxfBufSize
+    rxfConsumed' = rxfConsumed + consumed
+
+    -- Minimum window update size
+    threshold = rxfBufSize `div` 2
```

Comment:
Why don't you use `unsafeShiftR`?

Reply:
Changed this to be back at its original definition:

```haskell
    threshold = rxfBufSize `unsafeShiftR` 1
```